sim/hunter/pet.go

```diff
@@ -215,32 +216,43 @@ func (hp *HunterPet) ExecuteCustomRotation(sim *core.Simulation) {
 
 	target := hp.CurrentTarget
 
+	// using Cast() directly is very expensive, since cast failures are logged, involving string operations
+	tryCast := func(spell *core.Spell) bool {
+		if !spell.CanCast(sim, target) {
+			return false
+		}
+		if !spell.Cast(sim, target) {
+			panic(fmt.Sprintf("Cast failed after CanCast() for spell %d", spell.SpellID))
+		}
+		return true
+	}
```

Comment:
Do you have numbers on the performance comparison here? This changes all 3 hunter spec results - albeit by very small amounts - so I just want to make sure everything checks out